### PR TITLE
Feature/basic malcolm simulator #20

### PR DESCRIPTION
**Pull request by ray-millward-tessella**
Added the logic to the test socket server to return a canned response if the request matches (ignoring the id).
Also added handling for subscribing and unsubscribing.
Added a delay to simulate a slow response.
Added a demo script and a demo server to demonstrate and test the socket responses.

@thomascobb I'm not sure if you care about this kind of Malcolm related code but I've included you in case you want to see/comment.

connect to #20

### CODE REVIEW

File: demos/sprint0/20-dev-malcolm-server.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# 20 - Make a Malcolm socket simulator
+
+## Demo
+
+1. Start the dev server
+    ```
+    npm run server:dev
+    ```
+1. Start the interactive socket demo server
```
Comment (thomascobb) [inline demos/sprint0/20-dev-malcolm-server.md]: client?

Comment (ray-millward-tessella) [reply demos/sprint0/20-dev-malcolm-server.md]: no, there is a demo js file in the sprint0 folder that allows me to query the server from the command line - this is simply to prove that I've set up the test malcolm server correctly without having to spin up MalcolmJS

Comment (ray-millward-tessella) [reply demos/sprint0/20-dev-malcolm-server.md]: ah did you mean this should say "socket demo client"? Yes, probably. It's a little confusing as I run the demo app as a node server but read from the command line.

Comment (ray-millward-tessella) [reply demos/sprint0/20-dev-malcolm-server.md]: I'm planning to fix the wording in my next pull request with other demos.

**Pull request merged**